extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/static_resource/StaticResourceRecorder.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.vertx.http.runtime.static_resource;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.runtime.configuration.ConfigurationException;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.StaticHandler;
+
+@Recorder
+public class StaticResourceRecorder {
+    private static final Logger LOGGER = Logger.getLogger(StaticResourceRecorder.class.getName());
+
+    public Handler<RoutingContext> handler(StaticResourceBuildTimeConfig buildTimeConfig,
+            StaticResourceRuntimeConfig runtimeConfig, String rawEndpoint, String endpoint) {
+
+        StaticResourceBuildTimeConfig.StaticResourceGeneralConfig config = buildTimeConfig.config.get(rawEndpoint);
+        StaticResourceRuntimeConfig.StaticResourcePathConfig staticResourcePathConfig = runtimeConfig.paths.get(rawEndpoint);
+        if (staticResourcePathConfig == null) {
+            throw new ConfigurationException(String.format("Static file directory for endpoint %s was not set", rawEndpoint));
+        }
+
+        final Path staticResourceDir = Paths.get(staticResourcePathConfig.path);
+        if (Files.notExists(staticResourceDir)) {
+            LOGGER.warnf("Static file directory %s does not exist", staticResourceDir);
+        }
+
+        final StaticHandler staticHandler = StaticHandler.create()
+                .setAllowRootFileSystemAccess(true)
+                .setWebRoot(staticResourcePathConfig.path)
+                .setDirectoryListing(config.allowDirectoryListing)
+                .setCachingEnabled(config.enableCache)
+                .setMaxCacheSize(config.cacheSize.get().asBigInteger().intValueExact())
+                .setIndexPage(config.index.get())
+                .setDefaultContentEncoding("UTF-8");
+
+        return new Handler<RoutingContext>() {
+            @Override
+            public void handle(RoutingContext routingContext) {
+                if (routingContext.normalisedPath().equals(endpoint)) {
+                    routingContext.response().setStatusCode(302);
+                    routingContext.response().headers().set(HttpHeaders.LOCATION, endpoint + "/");
+                    routingContext.response().end();
+                    return;
+                }
+
+                final Path staticFilePath = staticResourceDir
+                        .resolve(routingContext.normalisedPath().replaceFirst(endpoint + "/", ""));
```

Comment:
I suspect this won't work very well on Windows?

Reply:
The `StaticResourceTest` test did not fail in the CI. I guess we should be okay for Windows too.